cocotb/decorators.py

```diff
@@ -91,16 +92,37 @@ def __init__(self, inst):
 
         if inspect.iscoroutine(inst):
             self._natively_awaitable = True
+
         elif inspect.isgenerator(inst):
             self._natively_awaitable = False
+
         elif sys.version_info >= (3, 6) and inspect.isasyncgen(inst):
             raise TypeError(
-                "{} is an async generator, not a coroutine. "
-                "You likely used the yield keyword instead of await.".format(
-                    inst.__qualname__))
+                "'{}' is an async generator, not a coroutine.\n"
+                "You likely used the `yield` keyword instead of `await`."
+                .format(inst.__qualname__))
+
+        elif isinstance(inst, cocotb.coroutine) or inspect.iscoroutinefunction(inst):
+            raise TypeError(
+                "'{}' is an unstarted coroutine function.\n"
+                "Did you forget to add parentheses to the @cocotb.test() decorator?\n"
+                "Or forget to call the coroutine function?"
+                .format(coroutine.__qualname__))
+
+        elif sys.version_info >= (3, 6) and inspect.isasyncgenfunction(inst):
+            raise TypeError(
+                "'{}' is an unstarted async generator function.\n"
+                "You likely used the `yield` keyword instead of `await`.\n"
+                "And you also forgot to call the function."
+                .format(inst.__qualname__))
+
         else:
             raise TypeError(
-                "%s isn't a valid coroutine! Did you forget to use the yield keyword?" % inst)
+                "'{}' isn't a valid cocotb coroutine.\n"
+                "Did you forget to decorate your function with `@cocotb.coroutine`?\n"
+                "And also forget a `yield` statement?"
```

Comment:
Perhaps worth just mentioning `async` / `await` here, instead of pointing users to `@cocotb.coroutine`.